### PR TITLE
freebsd: fix create_group() cmd

### PR DESCRIPTION
**Pull request by goneri**
The correct command to create a group if `pw group add foo`. In
addition, this commit simplify a bit the logic:

- rename the `name` variable `group`
- simplify a block to avoid an extra level of indentation

### CODE REVIEW

**Comment by igalic** [comment]
can we add a test for this?

**Comment by goneri** [comment]
The best way to test this kind of thing is an integration test. I found the issue with an ad-hoc `user_data file and a post-deployment validation script. Is there already an infrastructure in place to test  this kind of thing?

File: cloudinit/distros/freebsd.py [hunk 1]
```diff
@@ -59,30 +59,31 @@ def _read_hostname(self, filename, default=None):
     def _write_hostname(self, hostname, filename):
         rhel_util.update_sysconfig_file(filename, {'hostname': hostname})
 
-    def create_group(self, name, members):
-        group_add_cmd = ['pw', '-n', name]
-        if util.is_group(name):
-            LOG.warning("Skipping creation of existing group '%s'", name)
+    def create_group(self, group, members):
```
Comment (OddBloke) [inline cloudinit/distros/freebsd.py]: I would suggest not changing the signature of this method from its parent.

Nothing is broken by this currently, but it means that callers can't use keyword arguments (i.e. `create_group(name="...")` works across `Distro` classes today, but would stop working with this change). (Or, more likely, it means that if callers introduce code which uses a keyword argument, we would likely break FreeBSD without noticing.)

(Similarly, we might want to change `members` to have a default value, but that inconsistency isn't introduced by this PR so I don't think we need to do it here.)

Comment (igalic) [reply cloudinit/distros/freebsd.py]: ooooooof, that rename was my suggestion 😅

Comment (OddBloke) [reply cloudinit/distros/freebsd.py]: Oops!

I wouldn't be opposed to renaming the parameter in both places, the real problem is the inconsistency.

Comment (goneri) [reply cloudinit/distros/freebsd.py]: Yes, good point. I fixed the argument name.

File: cloudinit/distros/freebsd.py [hunk 2]
```diff
@@ -60,7 +60,7 @@ def _write_hostname(self, hostname, filename):
         rhel_util.update_sysconfig_file(filename, {'hostname': hostname})
 
     def create_group(self, name, members):
-        group_add_cmd = ['pw', '-n', name]
+        group_add_cmd = ['pw', 'group', 'add', name]
```
Comment (igalic) [inline cloudinit/distros/freebsd.py]: 👍

**Pull request merged**